src/types/infer.ts

```diff
@@ -2662,4 +2663,90 @@ export class InferType {
 
         return resolvedCalleeT;
     }
+
+    private isExternalCallContext(expr: Expression): boolean {
+        if (expr instanceof Identifier) {
+            const exprT = this.typeOf(expr);
+
+            if (exprT instanceof UserDefinedType) {
+                if (exprT.definition instanceof ContractDefinition) {
+                    return true;
+                }
+            }
+
+            if (exprT instanceof TypeNameType) {
+                return (
+                    exprT.type instanceof UserDefinedType &&
+                    exprT.type.definition instanceof ContractDefinition &&
+                    exprT.type.definition.kind === ContractKind.Library
+                );
+            }
+        }
+
+        if (
+            expr instanceof MemberAccess ||
+            expr instanceof FunctionCallOptions ||
+            expr instanceof FunctionCall
+        ) {
+            return this.isExternalCallContext(expr.vExpression);
```

Comment:
I think this code may break for code like this one:

```
import * as Mod from "mod.sol";

...
   Mod.Library.fun(...);
```

If we keep resolving the `vExpression` of `MemberAccess` eventually we will hit `Mod` instead of `Library`.

Reply:
Addressed in 4ea38d2. See extended if coniditon at the beginning of the function.